### PR TITLE
refine all github workflows

### PR DESCRIPTION
**Pull request by M00nF1sh**
**What type of PR is this?**
cleanup

**Which issue does this PR fix**:
N/A

**What does this PR do / Why do we need it**:
This PR doing a few clean ups to our Github workflows, mainly
* migrate off self-hosted runners and use github runner instead
* explicitly define permission for each workflow's github token.
Also, a few clean ups are done:
* use the latest action versions, such as checkout/v3
* refine workflow filenames/descriptions to follow similar pattern.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:
Tested on a forked repo: https://github.com/M00nF1sh/amazon-vpc-cni-k8s

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Checkout latest commit in the PR
+        uses: actions/checkout@v3
+      - name: Set up Docker QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.18"
```
Comment (jdn5126) [inline .github/workflows/integration-tests.yaml:25]: nit: 1.19?

File: .github/workflows/weekly-cron-tests.yaml [hunk 2]
```diff
@@ -1,94 +1,83 @@
...
-          MNG_ROLE_ARN: ${{ secrets.MNG_ROLE_ARN }}
-          RUN_PERFORMANCE_TESTS: true
           PERFORMANCE_TEST_S3_BUCKET_NAME: cni-performance-tests
-          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+          RUN_PERFORMANCE_TESTS: true
           RUN_TESTER_LB_ADDONS: true
-          S3_BUCKET_CREATE: false
-          S3_BUCKET_NAME: ${{ secrets.S3_BUCKET_NAME }}
         run: |
           ./scripts/run-integration-tests.sh
```
Comment (jdn5126) [inline .github/workflows/weekly-cron-tests.yaml:46]: This script is already set above to run nightly. Maybe we should run `run-cni-release-tests` nightly instead?

**Review by jayanthvn — DISMISSED** [review]
lgtm

File: .github/workflows/weekly-cron-tests.yaml (hunk 2 above)
Comment (jayanthvn) [reply .github/workflows/weekly-cron-tests.yaml:46]: This is the performance test - "RUN_PERFORMANCE_TESTS: true"

Comment (jayanthvn) [reply .github/workflows/weekly-cron-tests.yaml:46]: We run performance, bottlerocket and calico tests weekly.

Comment (jdn5126) [reply .github/workflows/weekly-cron-tests.yaml:46]: Ah, I see now that the env vars determine which tests run, thank you for clarifying. Resolving this.

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (M00nF1sh) [reply .github/workflows/integration-tests.yaml:25]: yeah, 1.19 seems a good fit for the workflows, i'll update it.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -3,30 +3,23 @@ name: VPC CNI Release
...
   GITHUB_USERNAME: ${{ secrets.EKS_BOT_GITHUB_USERNAME }}
   GITHUB_TOKEN: ${{ secrets.EKS_BOT_GITHUB_TOKEN }}
 
 jobs:
-
   release:
-    name: Release
-    runs-on: ubuntu-20.04
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    runs-on: ubuntu-20.04
```
Comment (orsenthil) [inline .github/workflows/release.yaml:12]: [minor] - ubuntu-latest could be used here as well.

Comment (M00nF1sh) [reply .github/workflows/release.yaml:12]: +1, nice catch

**Review by jayanthvn — DISMISSED** [review]
lgtm

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (M00nF1sh) [reply .github/workflows/integration-tests.yaml]: addressed :D

File: .github/workflows/release.yaml (hunk 3 above)
Comment (M00nF1sh) [reply .github/workflows/release.yaml]: addressed

**Review by jayanthvn — APPROVED** [review]

**Review by orsenthil — APPROVED** [review]

**Pull request merged**